config/src/main/scala/io/buoyant/config/JsonStreamParser.scala

```diff
@@ -73,12 +73,15 @@ class JsonStreamParser(mapper: ObjectMapper with ScalaObjectMapper) {
     // Wrap the reader so reads just terminate if a non-fatal exception occurs. This ensures the below AsyncStream ends
     // if reads fail
     val terminatingReader = new Reader[Buf] {
+      val closeP = Promise[StreamTermination]()
       def read(): Future[Option[Buf]] = reader.read().handle {
         case NonFatal(e) =>
           log.debug(e, "Error reading JSON stream")
           None
       }
       def discard(): Unit = reader.discard()
+
+      def onClose: Future[StreamTermination] = closeP
```

Comment:
this promise is never satisfied?  why shouldn't this be a pass through to `reader.onClose`?